File: app/src/main/java/com/main/climbingdiary/common/EnvironmentParamter.java
package com.main.climbingdiary.common;

import com.main.climbingdiary.activities.MainActivity;

import java.time.LocalDate;

public class EnvironmentParamter {
    public static final int DB_VERSION = 1;
    public static final String DB_NAME = "touren.db";
    public static final String DB_NAME_COPY = "touren_copy.db";
    public static final String dbExportName = String.format("%s_%s.db",
            EnvironmentParamter.DB_NAME.replace(".db", ""),
            LocalDate.now().toString().replace("/", "_"));
    public static final String PACKAGE_NAME = MainActivity.getMainAppContext().getPackageName();
    public static final String DB_PATH = "/data/" + EnvironmentParamter.PACKAGE_NAME + "/databases/" + DB_NAME;
    public static final String DB_PATH_COPY = "/data/" + EnvironmentParamter.PACKAGE_NAME + "/databases/" + DB_NAME_COPY;
}

File: app/src/main/java/com/main/climbingdiary/common/EnvironmentParamter.kt
package com.main.climbingdiary.common

import android.content.Context
import com.main.climbingdiary.activities.MainActivity
import java.time.LocalDate

object EnvironmentParamter {
    private val context: Context by lazy { MainActivity.getMainAppContext() }
    const val DB_VERSION = 1
    const val DB_NAME = "touren.db"
    private const val DB_NAME_COPY = "touren_copy.db"
    val dbExportName =
        "${DB_NAME.replace(".db", "")}_${LocalDate.now().toString().replace("/", "_")}.db"
    private val PACKAGE_NAME: String = context.packageName
    val DB_PATH = "/data/$PACKAGE_NAME/databases/$DB_NAME"
    val DB_PATH_COPY = "/data/${PACKAGE_NAME}/databases/${DB_NAME_COPY}"
}
